### PR TITLE
Implement Excuse Generator

### LINKED ISSUE #10: As a user, I want a command to generate an excuse not to do any IT work.
- **Given** a user wants to generate an excuse for something not working.
- **When** a user types in {prefix}get_excuse.
- **Then** get a random excuse from the [Bastard Operator From Hell](http://pages.cs.wisc.edu/~ballard/bofh/bofhserver.pl) and return it.

(This is more for me to learn how Discord bots work)

### PR DESCRIPTION
**Pull request by DonSagiv**

### CODE REVIEW

File: Code2Gether-Discord-Bot/Modules/UserModules/ExcuseModule.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using System.Threading.Tasks;
+
+namespace Code2Gether_Discord_Bot.Modules.UserModules
+{
+    class ExcuseModule : ModuleBase<SocketCommandContext>
+    {
+        [Command("get_excuse", RunMode = RunMode.Async)]
+        [Alias("getexcuse")]
+        [Summary("Gets a random excuse.")]
+        public async Task GetExcuseAsync() =>
```
Comment (nickmartin1ee7) [inline Code2Gether-Discord-Bot/Modules/UserModules/ExcuseModule.cs:12]: Permissions check should be here

File: Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            _logger.Log(_context);
+
+            Embed embed;
+
+            try
+            {
+                if (!_context.Guild.GetCurrentUserAsync().Result.GuildPermissions.ManageMessages)
+                {
+                    throw new Exception("Missing ManageMessages permission for Message Latency");
+                }
```
Comment (DonSagiv) [inline Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs]: Not sure if this only for Ping or for can be used for other stuff.

File: Code2Gether-Discord-Bot.Tests/ExcuseGeneratorTest.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        public void EmbedColorHasValueTest() =>
+            Assert.IsTrue(_logic.Execute().Color.HasValue);
+
+        [Test]
+        public void EmbedHasTitleTest() =>
+            Assert.IsTrue(_logic.Execute().Title.Length > 0);
+
+        [Test]
+        public void EmbedHasDescriptionTest() =>
+            Assert.IsTrue(_logic.Execute().Description.Length > 0);
```
Comment (DonSagiv) [inline Code2Gether-Discord-Bot.Tests/ExcuseGeneratorTest.cs:74]: These are copied from the ping test as well. I don't know if this truly reflects whether or not my logic works.

File: Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                    .WithTitle("Greetings. My excuse today is:")
+                    .WithDescription("I'm not working.")
+                    .WithAuthor(_context.Message.Author)
+                    .Build();
+            }
+
+            return embed;
+        }
+
+        public async Task<string> generateExcuseAsync()
```
Comment (nickmartin1ee7) [inline Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs]: Should be `private` and naming convention should be PascalCase

File: Code2Gether-Discord-Bot/Modules/UserModules/ExcuseModule.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using Code2Gether_Discord_Bot.Static;
+using Discord.Commands;
+using System.Threading.Tasks;
+
+namespace Code2Gether_Discord_Bot.Modules.UserModules
+{
+    class ExcuseModule : ModuleBase<SocketCommandContext>
+    {
+        [Command("get_excuse", RunMode = RunMode.Async)]
+        [Alias("getexcuse")]
```
Comment (DonSagiv) [inline Code2Gether-Discord-Bot/Modules/UserModules/ExcuseModule.cs:10]: Hopefully I'm using this properly.

File: Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                {
+                    throw new Exception("Missing ManageMessages permission for Message Latency");
+                }
+
+                var temporaryMessage = _context.Channel.SendMessageAsync("https://tenor.com/view/loading-buffering-gif-8820437").Result;
+
+                embed = new EmbedBuilder()
+                    .WithColor(Color.DarkBlue)
+                    .WithTitle("Greetings. My excuse today is:")
+                    .WithDescription("I just don't feel like it.")
```
Comment (nickmartin1ee7) [inline Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs]: Is this where the reference to the `generateExcuseAsync()` should be?

File: Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+        public Embed Execute()
+        {
+            _logger.Log(_context);
+
+            Embed embed;
+
+            try
+            {
+                if (!_context.Guild.GetCurrentUserAsync().Result.GuildPermissions.ManageMessages)
```
Comment (nickmartin1ee7) [inline Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs]: In this command, requiring the ManageMessages role for this to work isn't a good design choice. In the Ping command it was needed to calculate timestamps. You can simply omit the `temporaryMessage` completely

**Review by nickmartin1ee7 — CHANGES_REQUESTED** [review]

File: Code2Gether-Discord-Bot.Tests/ExcuseGeneratorTest.cs (hunk 3 above)
Comment (nickmartin1ee7) [reply Code2Gether-Discord-Bot.Tests/ExcuseGeneratorTest.cs:74]: I'll look into it

File: Code2Gether-Discord-Bot/Modules/UserModules/ExcuseModule.cs (hunk 1 above)
Comment (nickmartin1ee7) [reply Code2Gether-Discord-Bot/Modules/UserModules/ExcuseModule.cs:12]: Unless ManageMessages is omitted from logic

File: Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs (hunk 6 above)
Comment (nickmartin1ee7) [reply Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs]: Should be Color.Purple to match bot's profile picture theme.

**Review by nickmartin1ee7 — COMMENTED** [review]
This change looks good to me!
I'll look at it closer

File: Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        #endregion
+
+        public Embed Execute()
+        {
+            _logger.Log(_context);
+
+            var response = GenerateExcuse();
+
+            return new EmbedBuilder()
+                .WithColor(Color.DarkBlue)
```
Comment (nickmartin1ee7) [inline Code2Gether-Discord-Bot.Library/BusinessLogic/ExcuseGeneratorLogic.cs]: Should be Color.Purple to match bot's PFP

File: Code2Gether-Discord-Bot/Modules/UserModules/ExcuseModule.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
+﻿using Code2Gether_Discord_Bot.Static;
+using Discord.Commands;
+using System.Threading.Tasks;
+
+namespace Code2Gether_Discord_Bot.Modules.UserModules
+{
+    class ExcuseModule : ModuleBase<SocketCommandContext>
```
Comment (nickmartin1ee7) [inline Code2Gether-Discord-Bot/Modules/UserModules/ExcuseModule.cs]: Must be public for command to be registered

**Review by nickmartin1ee7 — CHANGES_REQUESTED** [review]
Protection modify not present makes command unusable.

**Review by nickmartin1ee7 — APPROVED** [review]
lgtm

**Pull request merged**